lib/traject/readers/purl_fetcher_reader.rb

```diff
@@ -0,0 +1,74 @@
+class Traject::PurlFetcherReader
+  attr_reader :input_stream, :settings
+
+  def initialize(input_stream, settings)
+    @settings = Traject::Indexer::Settings.new settings
+    @input_stream = input_stream
+  end
+
+  def each
+    return to_enum(:each) unless block_given?
+
+    changes(first_modified: first_modified).each do |change|
+      yield PublicXmlRecord.new(change['druid'].sub('druid:', ''))
+    end
+  end
+
+  private
+
+  def first_modified
+    settings['purl_fetcher.first_modified']
+  end
+
+  ##
+  # @return [Enumerator]
+  def changes(params = {})
+    paginated_get('/docs/changes', 'changes', params)
+  end
+
+  ##
+  # @return [Enumerator]
+  def deletes(params = {})
+    paginated_get('/docs/deletes', 'deletes', params)
+  end
+
+  ##
+  # @return [Hash] a parsed JSON hash
+  def get(path, params = {})
+    JSON.parse(HTTP.get("https://purl-fetcher-prod.stanford.edu" + path, params: params).body)
```

Comment:
If we go this way, this should probably be configurable in a way we don't yet have available to us.